### PR TITLE
ARGO-138 Added multi tenant functionality for SF reports and unit tests

### PR DESCRIPTION
**Pull request by pkoro**
- Added modifications in SF controller function to check for tenant API key
- Added unit tests for the following:
 - Check daily SF results
 - Check monthly SF results
 - Check authorization works

### CODE REVIEW

**Comment by pkoro** [comment]
please review /cc @dpavlos @kaggis @makistsantekidis

**Comment by makistsantekidis** [comment]
+1

**Comment by pkoro** [comment]
Examples:

Old GET call and XML response for monthly results:
```
/api/v1/service_flavor_availability?start_time=2015-06-01T00:00:00Z&end_time=2015-06-03T00:00:00Z&profile=ch.cern.sam.ROC_CRITICAL&granularity=monthly&flavor=SF1
```

```
 <root>
   <Profile name="ch.cern.sam.ROC_CRITICAL">
     <Site Site="S1">
       <Flavor Flavor="SF1">
         <Availability timestamp="2015-06" availability="99.99" reliability="99.99"></Availability>
       </Flavor>
     </Site>
     <Site Site="S2">
       <Flavor Flavor="SF1">
         <Availability timestamp="2015-06" availability="99.65" reliability="99.65"></Availability>
       </Flavor>
     </Site>
   </Profile>
 </root>
```

New API call and XML response for monthly results:
```
/api/v1/service_flavor_availability?start_time=2015-06-01T00:00:00Z&end_time=2015-06-03T00:00:00Z&job=EGI_Critical&granularity=monthly
```
```
 <root>
   <Job name="EGI_Critical">
     <SuperGroup name="S1">
       <Flavor Flavor="SF1">
         <Availability timestamp="2015-06" availability="99.99" reliability="99.99"></Availability>
       </Flavor>
       <Flavor Flavor="SF2">
         <Availability timestamp="2015-06" availability="88.88" reliability="88.88"></Availability>
       </Flavor>
     </SuperGroup>
     <SuperGroup name="S2">
       <Flavor Flavor="SF1">
         <Availability timestamp="2015-06" availability="94.44" reliability="94.44"></Availability>
       </Flavor>
       <Flavor Flavor="SF2">
         <Availability timestamp="2015-06" availability="99.65" reliability="99.65"></Availability>
       </Flavor>
     </SuperGroup>
   </Job>
 </root>
```

Similar differences on daily granularity results.

File: app/serviceFlavorAvailability/serviceFlavorAvailabilityModel.go [hunk 1]
```diff
@@ -48,41 +48,41 @@ type SF struct {
 	Availability []*Availability
 }
 
-type Site struct {
-	XMLName xml.Name `xml:"Site" json:"-"`
-	Site    string   `xml:"Site,attr" json:"Site"`
-	SF      []*SF
+type SuperGroup struct {
```
Comment (dpavlos) [inline app/serviceFlavorAvailability/serviceFlavorAvailabilityModel.go]: Missing godoc here the below.

File: app/serviceFlavorAvailability/serviceFlavorAvailabilityController.go [hunk 2]
```diff
@@ -28,11 +28,13 @@ package serviceFlavorAvailability
...
+
+	"github.com/argoeu/argo-web-api/utils/authentication"
 	"github.com/argoeu/argo-web-api/utils/caches"
 	"github.com/argoeu/argo-web-api/utils/config"
 	"github.com/argoeu/argo-web-api/utils/mongo"
-	"net/http"
-	"strings"
 )
 
 func List(r *http.Request, cfg config.Config) (int, http.Header, []byte, error) {
```
Comment (dpavlos) [inline app/serviceFlavorAvailability/serviceFlavorAvailabilityController.go]: godoc here

**Comment by dpavlos** [comment]
+1

**Pull request merged**